protostar/commands/test/cheatcodes/expect_events_cheatcode.py

```diff
@@ -0,0 +1,60 @@
+from typing import TYPE_CHECKING, Callable
+
+from protostar.commands.test.expected_event import ExpectedEvent
+from protostar.commands.test.starkware.cheatcode import Cheatcode
+from protostar.commands.test.test_environment_exceptions import (
+    ExpectedEventMissingException,
+)
+
+if TYPE_CHECKING:
+    from protostar.commands.test.starkware.forkable_starknet import ForkableStarknet
+    from protostar.commands.test.test_execution_environment import (
+        TestExecutionEnvironment,
+    )
+
+
+class ExpectEventsCheatcode(Cheatcode):
+    def __init__(
+        self,
+        syscall_dependencies: Cheatcode.SyscallDependencies,
+        starknet: "ForkableStarknet",
+        test_execution_environment: "TestExecutionEnvironment",
+    ):
+        super().__init__(syscall_dependencies)
+        self.starknet = starknet
+        self.test_execution_environment = test_execution_environment
+
+    @property
+    def name(self) -> str:
```

Comment:
this could be static, am i right?

Reply:
Cheatcodes must implement common interface (abstract class) in order to easily register them (`hint_locals[cheatcode.name] = cheatcode.build()`). Theoretically this property could be static, but there would be a problem with defining abstract Cheatcode class and I using metaclasses just to make propery static would be an overkill.

> “Metaclasses are deeper magic than 99% of users should ever worry about. If you wonder whether you need them, you don’t (the people who actually need them know with certainty that they need them, and don’t need an explanation about why).”

Reply:
Understandable - just nitpicking